datalad/core/local/run.py

```diff
@@ -285,7 +285,10 @@ def prepare_inputs(dset, inputs, extra_inputs=None):
         for res in _install_and_reglob(dset, gp):
             yield res
         for res in dset.get(gp.expand(full=True), on_failure="ignore"):
-            if res.get("state") == "absent":
+            if res.get("action") == "get" and \
+                    res.get("status") == "impossible" and \
+                    res.get("message") == "path does not exist":
+                # MIH why just a warning if given inputs are not valid?
```

Comment:
I don't spot any obvious conceptual reasons why we couldn't be stricter here, and I'd be fine with switching to an error.  However, quickly testing that out, we would need to rework things due to a non-obvious interaction with GlobbedPaths (shown by the resulting failure in `test_run_inputs_no_annex_repo`).  I'm going to open an issue about run's installation of missing subdatasets only working in a restricted set of cases (same issue mentioned elsewhere in this thread).  It looks like fixing that will require core change to the same part of GlobbedPaths, so I'll make sure to note that we should keep this "warning -> error" switch in mind.